source/dem/dem.cc

```diff
@@ -842,6 +861,18 @@ DEMSolver<dim>::solve()
         momentum,
         force);
 
+      // We have to update the positions of the points on boundary faces and
+      // their normal vectors here. The localize_contact class deletes the
+      // particle-wall contact candidate if it exists in the contact list. As a
+      // result, when we update the points on boundary faces and their normal
+      // vectors, localize_contact deletes it from the output of broad search
+      // and they are not updated in the contact force calculations.
+      if (parameters.grid_motion.motion_type !=
+          Parameters::Lagrangian::GridMotion<dim>::MotionType::none)
+        grid_motion_object
+          ->update_boundary_points_and_normal_vectors_in_contact_list(
+            pw_pairs_in_contact, updated_boundary_points_and_normal_vectors);
```

Comment:
The update is obliged to be done twice in the while boucle?

Reply:
No, these are two different functions. update_boundary_info_after_grid_motion updates the normal vectors and vertices. update_boundary_points_and_normal_vectors_in_contact_list applies these updated points in the particle-wall contact list.